File: frontend/src/api/task.ts
import { get, post } from './client'

export interface Task {
  id?: string
  style_profile_id: string
  user_input: string
  model: string
  size: string
  api_quality: string
  final_prompt: string
  status: 'pending' | 'processing' | 'done' | 'failed'
  result_image_url: string
  error_message: string
  retry_count: number
  progress: number
  created_by: string
  created_at: string
  updated_at: string
  source_image_urls: string[]
  image_count: number
  parent_task_id?: string
}

export interface TaskListResponse {
  items: Task[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export function createTask(styleProfileId: string, userInput: string, model?: string, size?: string, apiQuality?: string, sourceImages?: string[], imageCount?: number): Promise<Task> {
  return post<Task>('/tasks', {
    style_profile_id: styleProfileId,
    user_input: userInput,
    model: model || '',
    size: size || 'auto',
    api_quality: apiQuality || 'medium',
    source_images: sourceImages || [],
    image_count: imageCount || 1,
  })
}

export function getTask(id: string): Promise<Task> {
  return get<Task>(`/tasks/${id}`)
}

export function listTasks(params?: { createdBy?: string; status?: string; page?: number; pageSize?: number }): Promise<TaskListResponse> {
  const query = new URLSearchParams()
  if (params?.createdBy) query.set('created_by', params.createdBy)
  if (params?.status && params.status !== 'all') query.set('status', params.status)
  if (params?.page) query.set('page', String(params.page))
  if (params?.pageSize) query.set('page_size', String(params.pageSize))
  const queryString = query.toString()
  return get<TaskListResponse>(`/tasks${queryString ? `?${queryString}` : ''}`)
}

export function subscribeTask(taskId: string, onUpdate: (data: any) => void): () => void {
  const protocol = location.protocol === 'https:' ? 'wss:' : 'ws:'
  const ws = new WebSocket(`${protocol}//${location.host}/ws/tasks/${taskId}`)

  ws.onmessage = (event) => {
    try {
      const data = JSON.parse(event.data)
      onUpdate(data)
    } catch {
      // ignore parse errors
    }
  }

  return () => ws.close()
}

export function sendChatMessage(taskId: string, message: string): Promise<Task> {
  return post<Task>(`/tasks/${taskId}/chat`, { message })
}

export function getTaskConversation(taskId: string): Promise<Task[]> {
  return get<Task[]>(`/tasks/${taskId}/conversation`)
}

File: backend/internal/model/task.go
package model

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type TaskStatus string

const (
	TaskStatusPending    TaskStatus = "pending"
	TaskStatusProcessing TaskStatus = "processing"
	TaskStatusDone       TaskStatus = "done"
	TaskStatusFailed     TaskStatus = "failed"
)

type Task struct {
	ID               primitive.ObjectID `json:"id" bson:"_id,omitempty"`
	StyleProfileID   primitive.ObjectID `json:"style_profile_id" bson:"style_profile_id"`
	UserInput        string             `json:"user_input" bson:"user_input"`
	Model            string             `json:"model" bson:"model"`
	Size             string             `json:"size" bson:"size"`
	APIQuality       string             `json:"api_quality" bson:"api_quality"`
	ProjectID        string             `json:"project_id" bson:"project_id"`
	FinalPrompt      string             `json:"final_prompt" bson:"final_prompt"`
	Status           TaskStatus         `json:"status" bson:"status"`
	ResultImageURL   string             `json:"result_image_url" bson:"result_image_url"`
	ErrorMessage     string             `json:"error_message" bson:"error_message"`
	PlatformTaskID   string             `json:"platform_task_id" bson:"platform_task_id"`
	Progress         int                `json:"progress" bson:"progress"`
	RetryCount       int                `json:"retry_count" bson:"retry_count"`
	CreatedBy        string             `json:"created_by" bson:"created_by"`
	CreatedAt        time.Time          `json:"created_at" bson:"created_at"`
	UpdatedAt        time.Time          `json:"updated_at" bson:"updated_at"`
}
